app/services/proxy_status.rb

```diff
@@ -0,0 +1,55 @@
+class ProxyStatus
+  CONNECTION_ERRORS = [Errno::EINVAL, Errno::ECONNRESET, EOFError, Timeout::Error, Errno::ENOENT,
+                       Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, Net::ProtocolError]
+
+  def initialize(proxy, opts = {})
+    raise ::Foreman::Exception.new(N_('Must specify a Smart Proxy to use')) unless proxy.is_a?(SmartProxy)
```

Comment:
Kinda confused at this check? I don't think it's worth protecting anything here, but in any case if an object walks like a SmartProxy, swims like a SmartProxy and quacks like a SmartProxy, I [call that a SmartProxy](https://en.wikipedia.org/wiki/Duck_typing).  

What I'm trying to say is that even though I doubt anyone will call ProxyStatus.new with anything other than a SmartProxy, if a plugin has an object that behaves like a SmartProxy it should probably be allowed as well?